File: primitive_types/count_bits/python/count_bits.py
#!/usr/bin/env python


def count_bits(num):

    counter = 0
    if not isinstance(num, int) or num < 0:
        print("Unexcepted value of num: {}".format(num))
        return counter

    while num:
        counter += num & 1
        num = num >> 1
    return counter


if __name__ == "__main__":
    print(count_bits(15333333333442434))

File: primitive_types/count_bits/javascript/count_bits.js

var countBits = function(num) {

    var counter = 0;
    while (num > 0) {
        counter = counter + (num & 1);
        num = num >> 1;
    }

    return counter;
}

console.log(countBits(15))
console.log(countBits(1))
console.log(countBits(0))
console.log(countBits(-10))
